livesync_citadel/indico_livesync_citadel/backend.py

```diff
@@ -0,0 +1,111 @@
+
+        super(LiveSyncCitadelUploader, self).__init__(*args, **kwargs)
+
+        self.search_app = LiveSyncCitadelPlugin.settings.get('search_backend_url')
+        self.tika_server = LiveSyncCitadelPlugin.settings.get('tika_server')
+        self.endpoint_url = url_join(self.search_app, 'api/records')
+        self.headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Authorization': 'Bearer {}'.format(LiveSyncCitadelPlugin.settings.get('search_backend_token'))
+        }
+
+    def get_jsondata(self, obj):
+        # TODO: should "schemas/indico" be another setting?
+        if isinstance(obj, Event):
+            event_data = EventSchema().dump(obj)
+            event_data['$schema'] = url_join(self.search_app, 'schemas/indico/events_v1.0.0.json')
+            return event_data, EntryType.event
+        elif isinstance(obj, Contribution):
+            contribution_data = ContributionSchema().dump(obj)
+            contribution_data['$schema'] = url_join(self.search_app, 'schemas/indico/contributions_v1.0.0.json')
+            return contribution_data, EntryType.contribution
+        elif isinstance(obj, SubContribution):
+            subcontrib_data = SubContributionSchema().dump(obj)
+            subcontrib_data['$schema'] = url_join(self.search_app, 'schemas/indico/subcontributions_v1.0.0.json')
+            return subcontrib_data, EntryType.subcontribution
+        elif isinstance(obj, Attachment):
+            attachment_data = AttachmentSchema().dump(obj)
+            attachment_data['$schema'] = url_join(self.search_app, 'schemas/indico/attachments_v1.0.0.json')
+            return attachment_data, EntryType.attachment
+        elif isinstance(obj, EventNote):
+            note_data = EventNoteSchema().dump(obj)
+            note_data['$schema'] = url_join(self.search_app, 'schemas/indico/notes_v1.0.0.json')
+            return note_data, EntryType.note
+        else:
+            raise ValueError('unknown object ref: {}'.format(obj))
+
+    def upload_jsondata(self, jsondata, change_type, obj_id, entry_type):
+        response = None
+        if change_type & SimpleChange.created:
+            response = requests.post(self.endpoint_url, headers=self.headers, json=jsondata, verify=False)
+        else:
+            search_id = LiveSyncCitadelSearchAppIdMap.get_search_id(obj_id, entry_type)
+            if search_id:
+                if change_type & SimpleChange.updated:
+                    url = url_join(self.search_app, 'api/record/{}'.format(search_id))
+                    response = requests.put(url, headers=self.headers, json=jsondata, verify=False)
+                elif change_type & SimpleChange.deleted:
+                    url = url_join(self.search_app, 'api/record/{}'.format(search_id))
+                    response = requests.delete(url, headers=self.headers, json=jsondata, verify=False)
```

Comment:
`verify=False` is only for easier development

Reply:
If you install the CERN CA on your machine and add `export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt` to your shell config, this should not be needed ;)

Reply:
Of course I did that, but this would involve changing the containers configuration. I didn't want to do that :P this way it is easier for me

Reply:
And apart from that, the CA was not signed by the CERN Root CA. It was just a self-signed cert so it wouldn't help anyway.

Reply:
fair enough, i didn't think about containers. maybe use `verify=(not config.DEBUG)` since with containers it's probably more common to not have a proper cert (unless you can use plain http there?)

Reply:
actually, I don't need to use `verify` keyword anymore since for quite some time I have been using http rather than https